### DISCUSSION TITLE
Ability to call notFound() from layout

### DISCUSSION THREAD

**Discussion by imclint21** [vercel/next.js#57764] [Ideas]
### Goals

Ability to call notFound() from layout

### Background

I'm using SWR in my layout and I would like to call `notFound()` directly when a called resource is not found.

### Proposal

`notFound()` should works from layout too.

**Comment by imclint21** [comment]
From the layout its only give an error like this:

<img width="952" alt="image" src="https://github.com/vercel/next.js/assets/5221349/20aa02fd-7eda-4754-8c72-f9dc0b1ba3f9">

**Comment by imclint21** [comment]
Everybody dead here? :)

**Comment by prkagrawal** [comment]
Their is a [notFound method](https://github.com/vercel/next.js/blob/2e82ca81114f40c12dcdff52c0b264d46f29fc5f/packages/next/src/client/components/not-found.ts#L9C1-L9C1), it can be used with
`import { notFound } from 'next/navigation';`

Now it depends on what your use-case is and how you want to handle it.

**Reply by imclint21** [reply]
Yes I especially speak about this one, but it does not seems works from outside a page, it's just trigger an error.

**Comment by landsman** [comment]
Is it possible to do it like this, but it will produce a call of `page.tsx` anyway.
So 2x API request for example 🤦

**Comment by imclint21** [comment]
Personally I have a `ApiWrapper` that contains a `SWRConfig`, so I would like handle error 404 mistake and trigger `notFound` from here, but not from any pages.
